# UoB-ThesisCodes

# Overview
In this project, we aimed to discover new potential immune markers to eliminate resistance to immunotherapy in hepatocellular carcinoma (HCC). To do so, public GEO (Gene Expression Omnibus) microarray datasets were analyzed using unsupervised (i.e., t-SNE and PCA) and supervised ML methods (i.e., Boruta and Random Forest) in an effort to discover signature genes. To generate transcriptomics signatures, we conducted feature selection with univariate analysis and filtered differentially expressed genes between adjacent and normal tissues. We further applied multivariate feature selection to each dataset via Boruta algorithm. Then, the optimal gene set was obtained following an optimization procedure, which utilized a Random Forest (RF) classifier. On the other hand, gene expression profiles of three microarray datasets were deconvoluted by the MCP-counter, CIBERSORT, TIMER, EPIC and quanTIseq immune deconvolution methods to estimate cell contents in the samples. Potential associations between identified transcriptomics signatures and immune markers were then obtained via a pathway enrichment analysis.
